Updating deprecations list (#7329)
fishtown-analytics/docs.getdbt.com: website/docs/reference/deprecations.md

As dbt runs, it generates different categories of [events](/reference/events-logging), one of which is _deprecations_. Deprecations are a special type of warning that lets you know that there are problems in parts of your project that will result in breaking changes in a future version of dbt. It is important to resolve any deprecation warnings in your project before the changes are made.

> ## Check for deprecation warnings

Author: ```suggestion
## Identify deprecation warnings
```

> Idenitifying which deprecations impact your environment is the first step towards remediation. 

Author: ```suggestion
Finding deprecations that impact your code can be a daunting task when looking at the standard logs. Identifying them is the first step towards remediation. There are several methods for quickly locating deprecations and automatically remediating some of them.
```

1. Replace `dbt_modules` with `dbt_packages` in your `clean-targets` spec (and `.gitignore`).
2. Set `packages-install-path: dbt_modules` if you want to keep having packages installed in `dbt_modules`.

### PackageMaterializationOverrideDeprecation

> The behavior where installed packages could override built-in materializations without your explicit opt-in is deprecated. Setting the [`require_explicit_package_overrides_for_builtin_materializations` flag](/reference/global-configs/behavior-changes#package-override-for-built-in-materialization) to `false` in your `dbt_project.yml` allowed package that matched the name of a built-in materialization to be continue to be included in the search and resolution order.

Author: ```suggestion
The behavior where installed packages could override built-in materializations without your explicit opt-in is deprecated. Setting the [`require_explicit_package_overrides_for_builtin_materializations` flag](/reference/global-configs/behavior-changes#package-override-for-built-in-materialization) to `false` in your `dbt_project.yml` allowed packages that matched the name of a built-in materialization to continue to be included in the search and resolution order.
```

#### PackageMaterializationOverrideDeprecation warning resolution

Explicitly override built-in materializations, in favor of a materialization defined in a package, by reimplementing the built-in materialization in your root project and wrapping the package implementation.

> For example: 

Reviewer: Can we be more explicit? I'm not sure if this is correct @graciegoheen. Also do you add it anywhere or is there a particular place?
```suggestion
For example, add this to your snowflake package:
```

Reviewer 2: it's in your root project

Reviewer 2: https://docs.getdbt.com/reference/global-configs/behavior-changes#package-override-for-built-in-materialization

Author: Cited above

```jinja

{% materialization table, snowflake %}
    {{ return (package_name.materialization_table_snowflake()) }}
{% endmaterialization %}

```

> Then, remove the `require_explicit_package_overrides_for_builtin_materializations` flag from your `dbt_project.yml`.

Author: ```suggestion
Then, remove the `require_explicit_package_overrides_for_builtin_materializations: false` flag from your `dbt_project.yml`.
```

> Then, remove the `require_explicit_package_overrides_for_builtin_materializations: false` flag from your `dbt_project.yml`.

Reviewer: ```suggestion
Once you add the override to your package, you can remove the `require_explicit_package_overrides_for_builtin_materializations: false` flag from your `dbt_project.yml`.
```

Author: I think we need to be a little stronger than "you can" They have to remove it or they won't resolve the warning. how about
```suggestion
Once you've added the override for your package, remove the `require_explicit_package_overrides_for_builtin_materializations: false` flag from your `dbt_project.yml` to resolve the warning.
```

#### SourceFreshnessProjectHooksNotRun warning resolution

Set `source_freshness_run_project_hooks` to `true`. For instructions on skipping project hooks during a `dbt source freshness` invocation, check out the [behavior change documentation](/reference/global-configs/behavior-changes#project-hooks-with-source-freshness).

### UnexpectedJinjaBlockDeprecation

> If you have an orphaned Jinja block, you will receive a warning, and in a future version, dbt will stop supporting unexpected Jinja blocks. Previously, these orphaned Jinja blocks were silently ignored.

Reviewer 2: ```suggestion
If you have an unexpected Jinja block - an orphaned Jinja block or a Jinja block outside of a macro context - you will receive a warning, and in a future version, dbt will stop supporting unexpected Jinja blocks. Previously, these unexpected Jinja blocks were silently ignored.
```

<File name='macros/my_macro.sql'>

```sql

{% endmacro %} # orphaned endmacro jinja block

{% macro hello() %}
hello!
{% endmacro %}

```
</File>

> If you have an orphaned Jinja block, you will receive a warning, and in a future version, dbt will stop supporting unexpected Jinja blocks. Previously, these orphaned Jinja blocks were silently ignored.

Reviewer 2: ```suggestion
If you have an unexpected Jinja block, you will receive a warning, and in a future version, dbt will stop supporting unexpected Jinja blocks. Previously, these unexpected Jinja blocks were silently ignored.
```

<File name='macros/my_macro.sql'>

```sql

{% endmacro %} # orphaned endmacro jinja block

{% macro hello() %}
hello!
{% endmacro %}

```
</File>

#### UnexpectedJinjaBlockDeprecation warning resolution

> Delete the orphaned Jinja blocks.

Reviewer 2: ```suggestion
Delete the unexpected Jinja blocks.
```

#### SourceFreshnessProjectHooksNotRun warning resolution

Set `source_freshness_run_project_hooks` to `true`. For instructions on skipping project hooks during a `dbt source freshness` invocation, check out the [behavior change documentation](/reference/global-configs/behavior-changes#project-hooks-with-source-freshness).

### UnexpectedJinjaBlockDeprecation

If you have an unexpected Jinja block - an orphaned Jinja block or a Jinja block outside of a macro context - you will receive a warning, and in a future version, dbt will stop supporting unexpected Jinja blocks. Previously, these unexpected Jinja blocks were silently ignored.

<File name='macros/my_macro.sql'>

```sql

{% endmacro %} # orphaned endmacro jinja block

{% macro hello() %}
hello!
{% endmacro %}

```
</File>

> If you have an unexpected Jinja block, you will receive a warning, and in a future version, dbt will stop supporting unexpected Jinja blocks. Previously, these unexpected Jinja blocks were silently ignored.

Reviewer: This repeats the above, right? I think we can delete [lines 321 to 334](https://github.com/dbt-labs/docs.getdbt.com/pull/7329/files#diff-0e4a2f7e2e1b9d4f984c9833dcb896edd5dd9cd91898de0567da4d35cdbe8e7aR321-R334)

Author: Whoops! Good catch
